### PR TITLE
Improve specs and docs in Macro.underscore/camelize

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/macro.ex [hunk 1]
```diff
@@ -1815,12 +1815,13 @@ defmodule Macro do
   end
 
   @doc """
-  Converts the given atom or binary to underscore format.
+  Converts `module` to a string with underscore-slash format.
```
Comment (josevalim) [inline lib/elixir/lib/macro.ex:1818]: I don't think we should use `module`. If you say it is a module, I would expect it to be something previous defined with `defmodule`. Not only that, it also accept `:FooBar` or "FooBar", and none of them are modules. It is really an atom or a string.

Comment (eksperimental) [reply lib/elixir/lib/macro.ex:1818]: I went with `module_name` but then decided to go with module.

Should we go with `atom_or_string`?

Comment (josevalim) [reply lib/elixir/lib/macro.ex:1818]: > Should we go with atom_or_string?

Sounds good to me!

Comment (eksperimental) [reply lib/elixir/lib/macro.ex]: Please have a look at the changes.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**